python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+        Debug variables
+    This task has no debug variables
+
+    \section ap_association_AssociationDBSqliteTask_Example
+        Example of using AssociationDBSqliteTask
+    This task has no standalone example, however it is applied as a subtask of
+    ap.association.AssociationTask
+    """
+
+    ConfigClass = AssociationDBSqliteConfig
+    _DefaultName = "association_db_sqlite"
+
+    def __init__(self, **kwargs):
+        """ Create a new connection to the sqlite database specified in
+        the config.
+        """
+        pipeBase.Task.__init__(self, **kwargs)
+        self.indexer = IndexerRegistry[self.config.indexer.name](
+            self.config.indexer.active)
+        self._db_connection = sqlite3.connect(self.config.db_name)
+        self._db_cursor = self._db_connection.cursor()
+
+    def run(self):
+        """ Not implemented in this subtask. Please use the store and load
+        methods.
+        """
+        Raise(NotImplementedError)
+
+    def commit(self):
+        """ Save changes to the sqlite database.
+        """
+        self._db_connection.commit()
+
+    def close(self):
+        """ Close the connection to the sqlite database.
+        """
+        self._db_connection.close()
+
+    def create_tables(self):
+        """ If no sqlite database exists with the correct tables we can create
+        one using this method.
+
+        Returns
+        -------
+        bool
+            Successfully created a new database with specified tables.
+        """
+
+        obj_schema = make_minimal_dia_object_schema()
+        src_schema = make_minimal_dia_source_schema()
+
+        self._db_cursor.execute(
+            'select name from sqlite_master where type = "table"')
+        db_tables = self._db_cursor.fetchall()
+
+        if db_tables:
+            return False
+        else:
+            name_type_string = ""
+            for sub_schema in obj_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects (%s)" % name_type_string)
+
+            name_type_string = ""
+            for sub_schema in src_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_sources (%s)" % name_type_string)
```

Comment:
You might look at `daf_ingest`, which is doing a lot of this (but in MySQL-ish SQL), but don't spend a lot of time on this.  In the not-too-distant future, the Butler should do this for you.  (Also, Tasks are not supposed to do any I/O, so this will eventually have to be refactored anyway.)

Reply:
Thanks for the heads up @ktlim. I already have a ticket to change this to something that does its i/o through the butler (DM-11766), once we start thinking about source association beyond ap_verify and the butler refactor becomes more concrete.

Reply:
@ktlim Do you mean that this should not be a Task at all, or that it should delegate I/O but can still be used to, e.g., construct queries?

Reply:
In the long run, the Task should, I think, take in a catalog of DIASources and a catalog of DIAObjects from the same general spatial area and do whatever association is necessary, returning a new catalog of DIAObjects.  The task of constructing database queries to retrieve and write the DIAObjects should ideally be left to the SuperTask (for dataId generation) and the Butler (for actual SQL).  But if performance proves to be a problem with that design, alternatives can be considered.